contracts/other-contracts/opensea-wrapper/OpenSeaWrapperFactory.sol

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.22;
+
+import { OpenSeaWrapper } from './OpenSeaWrapper.sol';
+import { IBosonVoucher } from '../../protocol-contracts/contracts/interfaces/clients/IBosonVoucher.sol';
+import { IBosonSellerHandler } from '../../protocol-contracts/contracts/interfaces/handlers/IBosonSellerHandler.sol';
+import { BosonTypes } from "../../protocol-contracts/contracts/domain/BosonTypes.sol";
+
+contract OpenSeaWrapperFactory {
+    address private protocolAddress;
+    address private unwrapperAddress;
+    address private wethAddress;
+    address private seaport;
+    uint256 private openSeaFee;
+    address payable openSeaRecipient;
+    bytes32 private openSeaConduitKey;
+    address private openSeaConduit;
+    mapping (address => address) private wrappers;
+
+    constructor(
+        address _protocolAddress,
+        address _wethAddress,
+        address _unwrapperAddress,
+        address _seaport,
+        uint256 _openSeaFee,
+        address payable _openSeaRecipient,
+        bytes32 _openSeaConduitKey,
+        address _openSeaConduit
+    ) {
+        protocolAddress = _protocolAddress;
+        wethAddress = _wethAddress;
+        unwrapperAddress = _unwrapperAddress;
+        seaport = _seaport;
+        openSeaFee = _openSeaFee;
+        openSeaRecipient = _openSeaRecipient;
+        openSeaConduitKey = _openSeaConduitKey;
+        openSeaConduit = _openSeaConduit;
+    }
+
+    function create(address _voucherContract) external returns (address) {
+        // Check no wrapper exist for this voucher contract
+        require(wrappers[_voucherContract] == address(0), "Voucher is already wrapped");
+        // Check the sender is the seller's assistant or admin wallet
```

Comment:
Technically, anyone could deploy this wrapper, since being a deployer does not give any additional privileges.

Especially if we think in a context of sequential commit, the owners of the rNFT should be able to do it themselves and not wait on the seller.

If we make the wrapper contract ownable (probably we will, so they can manage then on OS), it should just copy the bosonVoucer.owner(), regardless of who deploys the wrapper contract.

Reply:
Seller constraint is now removed. Everyone is able to create the Wrapper contract. 
Of course, only the token owner (usually the seller, then) is allowed to wrap it